api/api.go

```diff
@@ -9,24 +12,32 @@ import (
 type API struct {
 	db        *database.ServiceDB
 	container *container.Container
+	cfg       *config.Config
 }
 
 // Option is a configuration func for MESG.
 type Option func(*API)
 
 // New creates a new API with given options.
-func New(db *database.ServiceDB, options ...Option) (*API, error) {
-	a := &API{db: db}
+func New(options ...Option) (*API, error) {
```

Comment:
The API should be in parameter here, it's a required arguments so it has to be here. If we call `New()` without any options (what most developers will do without knowing the codebase) we will have some crash in the code.

Of course you add the error in the end but what's the point to remove a required argument and add a error, it will just be more confusing. I like the pattern to use options but for what is optional and not the stuff required, it doesn't make sense.

Also and more important, let's keep the pull requests simple. This has nothing to do here, it's a pull request about extending the API why do we have some stuff in the database. Especially that this modifications has been done in another pull request that was just merged today and stayed open for more than a week so for this kind of things it better to talk about this in the other PR and not try to include that in a non related PR.